doc/contribute/bin_blobs.rst

```diff
@@ -0,0 +1,201 @@
+as part of the submission process.
+
+Each blob which may be fetched must be individually identified in the
+corresponding :file:`module.yml` file. A specification for a blob must contain:
+
+- Purpose: an abstract description of the blob itself
+- A reference to vendor-provided documentation
+- The blob’s type, which must be one of the allowed types
+- A checksum for the blob, which the west extension checks after downloading.
+  This is required for reproducibility and to allow bisecting issues as blobs
+  change using Git and west
+- License text applicable to the blob or a reference to such text
+
+The west extension used to fetch blobs accepts command line arguments that
+allow the user to select the module or modules to fetch blobs from. This is
+required for user control and a uniform user experience regardless of the blob
+vendor.
+
+Allowed types
+=============
+
+The following binary blob types are acceptable in Zephyr:
+
+* Precompiled libraries: Hardware enablement libraries, distributed in
+  precompiled binary form, typically for SoC peripherals. An example could be an
+  enablement library for a wireless peripheral
+* Firmware images: An image containing the executable code for a secondary
+  processor or CPU.  This can be full or partial (typically delta or patch data)
+  and is generally copied into RAM or flash memory by the main CPU. An example
+  could be the firmware for the core running a Bluetooth LE Controller
+* Miscellaneous binary data files. An example could be pre-trained neural
+  network model data
+
+Note that just because a blob has an acceptable type does not imply that it will
+be unconditionally accepted by the project; any blob may be rejected for other
+reasons on a case by case basis.
+In case of disagreement, the TSC is the arbiter of whether a particular blob
+fits in one of the above types.
+
+Hardware agnostic features provided via a proprietary library are not
+acceptable. For example, a proprietary and hardware agnostic TCP/IP stack
+distributed as a static archive would be rejected.
+
+Precompiled library-specific requirements
+=========================================
+
+This section contains additional requirements specific to precompiled library
+blobs.
+
+Any person who wishes to submit a precompiled library must represent that it
+meets these requirements. The project may remove a blob from the upstream
+distribution if it is discovered that the blob fails to meet these requirements
+later on.
+
+Interface header files
+----------------------
+
+The precompiled library must be accompanied by one or more header files,
+distributed under an OSI approved license, that define the interface to the
+library.
+
+Allowed dependencies
+--------------------
+
+This section defines requirements related to external symbols that a library
+blob requires the build system to provide.
+
+* The blob must not depend on Zephyr APIs directly. This is required for loose
```

Comment:
Maybe this could be more precise. How about
```suggestion
* The blob must not depend on Zephyr APIs directly. In other words, it must have been possible to build the binary without any Zephyr source code or part of present. This is required for loose
```

Reply:
This could leave a `dlopen()/dlsym()` "loophole" potentially open but that would seem a bit "extreme" and... spirit/letter of the "law"?